src/directives/rcdni.js

```diff
@@ -0,0 +1,42 @@
+directives.directive('rcDni',['dniSemanticsValidationService','dniSyntaxValidationService',function(dniSemanticsValidationService,dniSyntaxValidationService){
+	return{
+		require: 'ngModel',					
+		link: function (scope, elm, attrs, ctrl){	
+			var validationToExecute = function (viewValue){
+				var currentTypedValue = viewValue;
+				validateDNI(currentTypedValue);
+			};
+
+			function validateDNI(dni){
+				if (validateDNISyntax(dni)){
```

Comment:
I agree with you, the method can be refactor to:

function validateDNI(dni){
return validateDNISyntax(dni) && validateDNISemantic(dni);
}